server/webapp/WEB-INF/rails.new/app/assets/javascripts/pluggable_scm_check_connection.js

```diff
@@ -0,0 +1,42 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2014 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END**********************************/
+
+PluggableSCMCheckConnection = function (url) {
+
+    _bind = function (form_selector, check_connection_selector, connection_message_selector) {
+        jQuery(form_selector).on('click', check_connection_selector, function (e) {
+            var connection_message_element = jQuery(connection_message_selector);
+            connection_message_element.removeClass("error_message").removeClass("ok_message").text('Checking connection...');
+            jQuery.ajax({
+                type: "POST",
+                url: url,
+                data: jQuery(form_selector).serialize(),
+                success: function (data) {
```

Comment:
Just like "success", the [error event](http://api.jquery.com/Ajax_Events/) will also need to be handled, and a message shown. Otherwise, if there's an error (server not accessible, internal server error, etc), it will get stuck at "Checking connection ...".